### PR TITLE
[IJ Plugin] Add inspection for @oneOf input type constructor invocation

### PR DESCRIPTION
**Pull request by BoD**
Screenshot 2023-11-23 at 20 11 47

----
Note: this also changes the return types of `Optional.absent()` and `Optional.present()`.

### CODE REVIEW

File: libraries/apollo-api/src/commonTest/kotlin/test/OptionalTest.kt [hunk 1]
```diff
@@ -37,7 +37,7 @@ class OptionalTest {
 
   @Test
   fun getOrThrowAbsent() {
-    val optional = Optional.absent<String?>()
+    val optional: Optional<String?> = Optional.absent()
```
Comment (BoD) [inline libraries/apollo-api/src/commonTest/kotlin/test/OptionalTest.kt:40]: Inference wasn't working here, I guess that means the `absent()` change is source-breaking in some cases :(

Comment (BoD) [reply libraries/apollo-api/src/commonTest/kotlin/test/OptionalTest.kt:40]: Yes it makes more sense like this ✅

File: intellij-plugin/src/main/resources/inspectionDescriptions/ApolloOneOfInputCreation.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<html>
+<body>
+Reports invalid constructor invocations of <code>@oneOf</code> input types.
+<p>
+    Exactly one field must be set, and it must be <code>Present</code>.
```
Comment (martinbonnin) [inline intellij-plugin/src/main/resources/inspectionDescriptions/ApolloOneOfInputCreation.html:5]:
```suggestion
    Exactly one input field must be set, and it must be <code>Present</code>.
```

Comment (martinbonnin) [reply intellij-plugin/src/main/resources/inspectionDescriptions/ApolloOneOfInputCreation.html:5]: Or maybe more like so?

```suggestion
    Constructor of `@oneOf` input class must have exactly one <code>Present</code> argument.
```

* Field is JVM/GraphQL. 
* Property is Kotlin
* Constructor takes arguments?

**Review by martinbonnin — APPROVED** [review]
A bit of naming bikeshedding but LGTM!

File: intellij-plugin/src/main/resources/inspectionDescriptions/ApolloOneOfInputCreation.html (hunk 2 above)
Comment (BoD) [reply intellij-plugin/src/main/resources/inspectionDescriptions/ApolloOneOfInputCreation.html:5]: Yup, much better!

File: intellij-plugin/src/main/resources/messages/ApolloBundle.properties [hunk 3]
```diff
@@ -142,6 +142,10 @@ inspection.endpointNotConfigured.displayName=GraphQL endpoint not configured
 inspection.endpointNotConfigured.reportText=GraphQL endpoint not configured
 inspection.endpointNotConfigured.quickFix=Add introspection block
 
+inspection.oneOfInputCreation.displayName=OneOf Input Object creation issue
+inspection.oneOfInputCreation.reportText.wrongNumberOfArgs=<html><tt>@oneOf</tt> input must have exactly one field set
```
Comment (BoD) [inline intellij-plugin/src/main/resources/messages/ApolloBundle.properties:146]:
```suggestion
inspection.oneOfInputCreation.reportText.wrongNumberOfArgs=<html><tt>@oneOf</tt> input class constructor must have exactly one argument
```

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**